test/sql/window/test_streaming_window.test

```diff
@@ -0,0 +1,119 @@
+# name: test/sql/window/test_streaming_window.test
+# description: Streaming window functions
+# group: [window]
+
+statement ok
+PRAGMA enable_verification
+
+statement ok
+PRAGMA explain_output = PHYSICAL_ONLY;
+
+statement ok
+create table integers (i int, j int)
+
+statement ok
+insert into integers values (2, 2), (2, 1), (1, 2), (1, NULL)
+
+# Test each of the streaming window functions
+query TT
+explain select row_number() over (), i, j from integers
+----
+physical_plan	<REGEX>:.*STREAMING_WINDOW.*
+
+query TTT
+select row_number() over (), i, j from integers
+----
+1	2	2
+2	2	1
+3	1	2
+4	1	NULL
+
+query TT
+explain select rank() over (), i, j from integers
+----
+physical_plan	<REGEX>:.*STREAMING_WINDOW.*
+
+query TTT
+select rank() over (), i, j from integers
+----
+1	2	2
+1	2	1
+1	1	2
+1	1	NULL
+
+query TT
+explain select dense_rank() over (), i, j from integers
+----
+physical_plan	<REGEX>:.*STREAMING_WINDOW.*
+
+query TTT
+select dense_rank() over (), i, j from integers
+----
+1	2	2
+1	2	1
+1	1	2
+1	1	NULL
+
+query TT
+explain select percent_rank() over (), i, j from integers
+----
+physical_plan	<REGEX>:.*STREAMING_WINDOW.*
+
+query TTT
+select percent_rank() over (), i, j from integers
+----
+0.0	2	2
+0.0	2	1
+0.0	1	2
+0.0	1	NULL
+
+# Some combinations of compatible window functions
+query TT
+explain select i, j, first_value(i) over (), first_value(j) over () from integers
+----
+physical_plan	<REGEX>:.*STREAMING_WINDOW.*
+
+query TTTT
+select i, j, first_value(i) over (), first_value(j) over () from integers
+----
+2	2	2	2
+2	1	2	2
+1	2	2	2
+1	NULL	2	2
+
+query TTT
+select row_number() over (), first_value(i) over (), first_value(j) over () from integers
+----
+1	2	2
+2	2	2
+3	2	2
+4	2	2
+
+query TT
+select row_number() over (), row_number() over () from integers
+----
+1	1
+2	2
+3	3
+4	4
+
+# Ignore nulls is not supported for streaming windows
+query TT
+explain select first_value(i IGNORE NULLS) over () from integers
+----
+physical_plan	<!REGEX>:.*STREAMING_WINDOW.*
+
+# When we combine a streaming and non-streaming window function over the same window,
+# We get a non-streaming window operator
+query TT
+explain select first_value(i) over (), last_value(i) over () from integers
+----
+physical_plan	<!REGEX>:.*STREAMING_WINDOW.*
+
+# When we combine a streaming and non-streaming window function over a different window,
+# We get both a streaming and a non-streaming window operator
+# The order between the two window operators is not important
+query TT
+explain select first_value(i) over (), last_value(i) over (order by j) from integers
+----
+physical_plan	<REGEX>:.*STREAMING_WINDOW.*WINDOW.*|.*WINDOW.*STREAMING_WINDOW.*
```

Comment:
I think it would make the most sense to put the streaming operator last. That way the non-streaming has less data to accumulate and sort?

Reply:
Agreed